### ISSUE TITLE
gatsby-remark-prismjs line highlight breaks when react-live is included

### ISSUE THREAD

**Issue by bvaughn** [gatsbyjs/gatsby#1486]
I recently noticed a bug with the `gatsby-remark-prismjs` plugin on a site I'm working on: code line-highlights show up when I navigate to a page initially, but if I reload the page in place they're missing. (There's no wrapper `<span class="gatsby-highlight-code-line">...</span>` after a page reload).

After a lot of trimming down, I've been able to demonstrate this bug in a small repro (at [bvaughn/gatsby-remark-prismjs-react-live-bug](https://github.com/bvaughn/gatsby-remark-prismjs-react-live-bug)). The tl;dr is that including any [`react-live`](https://github.com/FormidableLabs/react-live) component (even if it's not used) will break the `gatsby-remark-prismjs` plugin line highlight.

I will attempt to further track this issue down. I'm also unclear about whether I should file this as an issue for Gatsby or React Live, although I believe it's probably more appropriate to file it against React Live. That being said, for now I'm going to try filing against both in case either project author has ideas or suggestions.

Relates to FormidableLabs/react-live/issues/22

**Comment by bvaughn** [comment]
It looks like Gatsby is initially rendering the markup, with line highlights, and then `react-live` is requiring its own copy of `prism-core` which is clobbering Gatsby's initial content.

I can prevent this by updating Prism to check an element's contents before potentially re-highlighting it, but... I kind of doubt Prism would be interested in accepting that change.

**Comment by bvaughn** [comment]
Maybe the real fix here is for Gatsby to make use of Prism's `data-line` attribute as mentioned in [the Prism docs](http://prismjs.com/plugins/line-highlight/#how-to-use).

**Comment by bvaughn** [comment]
Looks like maybe this can be avoided by removing the `language` attribute from Gatsby's pre-generated `<pre>` tags. This way, requiring Prism a second time won't re-highlight the already-processed content.

**Comment by jbolda** [comment]
Would having our prism prefix the classes help?
http://prismjs.com/plugins/custom-class/

**Comment by bvaughn** [comment]
Not sure I understand. Would that prevent Prism from re-processing the already-processed markup? Any benefits for that vs the data attribute approach?

**Comment by bvaughn** [comment]
(From our conversation in Discord)

> **brianvaughn**: Hm. So looking at the Prism docs, I'm not so sure this does what you're thinking.
**brianvaughn**: http://prismjs.com/plugins/custom-class/#example
**brianvaughn**: The prefix isn't for the `<pre class="...">` but rather its nodes within
**brianvaughn**: Which wouldn't help with what I'm trying to address

**Comment by bvaughn** [comment]
This is super weird. After the change from #1491 was released, I upgrade the docs site and while the initial markup seems as expected:
```html
<pre class="gatsby-language-js">
  <code>
```

The markup after reloading is incorrect:
```html
<pre class="gatsby- language-js">
  <code class=" language-js">
```

However [the repro case](https://github.com/bvaughn/gatsby-remark-prismjs-react-live-bug) I provided does not illustrate this broken behavior. I am not currently sure why this is. ☹️ More investigation is required.
